### PR TITLE
Add link types and metadata 

### PR DESCRIPTION
**Pull request by diegoceccarelli**

### CODE REVIEW

**Comment by diegoceccarelli** [comment]
I made a test on my server, It took around 10 hours to convert a full dump, and now the dump is 17Gb compressed (vs 15gb with the normal links). I think it's fine.

**Comment by tgalery** [comment]
I'm just wondering if we should squash some of the commits (I might try to backport some of these changes to the idio repo). I'm happy to sqash some of mine and @rod3go 's commits. If you are ok with that should I push to this branch straight ?

**Comment by diegoceccarelli** [comment]
I think that when you merge you can optionally ask to squash the commit in
one unique one. I would do that..

File: src/main/java/it/cnr/isti/hpc/wikipedia/parser/ArticleParser.java [hunk 1]
```diff
@@ -149,11 +138,100 @@ private void setWikiTitle(Article article) {
...
+				item++;
+			}
+			listId++;
+		}
+	}
+
+
+
+	private void setLinks(Article article, ParsedPage page){
+		final List<Link> links = new ArrayList<Link>(page.getLinks().size());
```
Comment (tgalery) [inline src/main/java/it/cnr/isti/hpc/wikipedia/parser/ArticleParser.java]: weird indentation

**Comment by diegoceccarelli** [comment]
@tgalery I indented the code (it was weird indeed) and squashed everything in one unique commit

**Comment by tgalery** [comment]
@diegoceccarelli looks good to me. I'm happy for it to be merged.

**Comment by diegoceccarelli** [comment]
Go for it! :)

**Pull request merged**